zeeguu_api/api/teacher_dashboard.py

```diff
@@ -272,6 +271,20 @@ def _link_teacher_cohort(user_id, cohort_id):
     db.session.commit()
     return 'added teacher_cohort relationship'
 
+@api.route("/users_by_teacher/<duration>", methods=["GET"])
+@with_session 
+def users_by_teacher(duration):
+    '''
+        Return list of dictionaries containing user info for all cohorts that the logged in user owns.
+
+    '''
+    from zeeguu_core.model import TeacherCohortMap
+    mappings = TeacherCohortMap.query.filter_by(user_id=flask.g.user.id).all()
+    all_users = []
+    for m in mappings:
+        users = users_from_cohort(m.cohort_id, duration)
+        all_users.extend(json.loads(users))
```

Comment:
i'm a bit puzzled by this. 

are you sure that converting to json and then deconverting 
again here is the nicest way to solve the problem you have?
i wonder if you could refactor the other function to extract a local 
that you can reuse both here and there ... something like: 

```
user_info_from_cohort
        c = Cohort.query.filter_by(id=id).one()
        users = User.query.filter_by(cohort_id=c.id).all()
        users_info = []
        for u in users:
            info = _get_user_info_for_teacher_dashboard(u.id, duration)
            users_info.append(info)
       return users_info
```

Reply:
Good point. much cleaner now :)